--- source/tools/toolutil/toolutil.cpp	2012-04-06 00:46:28 +0400
+++ source/tools/toolutil/toolutil.cpp	2012-05-12 09:32:21 +0400
@@ -40,9 +40,11 @@
 #   include <sys/types.h>
 #endif
 
-/* In MinGW environment, io.h needs to be included for _mkdir() */
+/* In MinGW environment, io.h needs to be included for _mkdir() for
+ * the mingw.org toolchain and direct.h for the mingw-w64 toolchain */
 #if U_PLATFORM == U_PF_MINGW
 #include <io.h>
+#include <direct.h>
 #endif
 
 #include <errno.h>
