--- work.orig/libarchive/archive_openssl_evp_private.h	2017-07-09 21:38:04.000000000 -0500
+++ work/libarchive/archive_openssl_evp_private.h	2018-04-22 08:25:29.735929400 -0500
@@ -31,6 +31,11 @@
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 #include <stdlib.h> /* malloc, free */
 #include <string.h> /* memset */
+
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
+#define inline __inline
+#endif
+
 static inline EVP_MD_CTX *EVP_MD_CTX_new(void)
 {
 	EVP_MD_CTX *ctx = (EVP_MD_CTX *)calloc(1, sizeof(EVP_MD_CTX));
--- work.orig/libarchive/archive_openssl_hmac_private.h	2017-07-09 21:38:04.000000000 -0500
+++ work/libarchive/archive_openssl_hmac_private.h	2018-04-22 08:37:10.860250700 -0500
@@ -31,6 +31,11 @@
 #if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 #include <stdlib.h> /* malloc, free */
 #include <string.h> /* memset */
+
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
+#define inline __inline
+#endif
+
 static inline HMAC_CTX *HMAC_CTX_new(void)
 {
 	HMAC_CTX *ctx = (HMAC_CTX *)calloc(1, sizeof(HMAC_CTX));
--- work.orig/libarchive/xxhash.c	2018-04-23 10:39:36.548437500 -0500
+++ work/libarchive/xxhash.c	2018-04-23 10:40:26.751993100 -0500
@@ -37,6 +37,10 @@
 #include "archive_xxhash.h"
 
 #ifdef HAVE_LIBLZ4
+
+#if defined(_MSC_VER) && (_MSC_VER < 1900)
+#define inline __inline
+#endif
 
 /***************************************
 ** Tuning parameters
