From fd5f8106a58601a963ee816e6a57aa7c61fafc53 Mon Sep 17 00:00:00 2001
From: BadWolf <b7f7@gmx.de>
Date: Sat, 6 Apr 2024 00:05:58 +0200
Subject: [PATCH] fix missing include folder for resource compiler

---
 build/cmake/lib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/cmake/lib/CMakeLists.txt b/build/cmake/lib/CMakeLists.txt
index 5d514ccb2e..43b14d1753 100644
--- a/build/cmake/lib/CMakeLists.txt
+++ b/build/cmake/lib/CMakeLists.txt
@@ -118,7 +118,7 @@ endmacro ()
 
 # Define directories containing the library's public headers
 set(PUBLIC_INCLUDE_DIRS ${LIBRARY_DIR})
-
+set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} /I ${LIBRARY_DIR}")
 # Split project to static and shared libraries build
 set(library_targets)
 if (ZSTD_BUILD_SHARED)
